streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -639,7 +639,31 @@ public void shouldRespectCommitNeeded() {
         assertTrue(task.commitNeeded());
 
         task.commit();
-        assertFalse(task.commitNeeded());
+        assertFalse(task.commitNeeded()); // confirm that timestamp was correctly committed
+        assertTrue(Long.parseLong(task.consumer.committed(partition1).metadata())
+                      == task.getPartitionTime(partition1));
+    }
+
+    @Test
+    public void shouldUpdateTimestampAfterRestart() {
```

Comment:
nit: `shouldRestorePartitionTimeOnRestart` -- "update timestamp" is a little fuzzy